BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -108,6 +125,213 @@ public boolean onOptionsItemSelected(MenuItem item) {
+      }
+
+      File file = new File(rootPath + fileName + ".bib");
+      if (file.exists()) {
+        file.delete();
+      }
+      file.createNewFile();
+
+      FileOutputStream out = new FileOutputStream(file);
+      out.flush();
+      out.close();
+
+      Toast.makeText(getContext(), getString(R.string.exported_file_stored_in) + '\n'
+          + "/" + folderName + "/" + fileName + ".bib", Toast.LENGTH_LONG).show();
+
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void retrieveBibContent() {
+
+    DatabaseHelper dbHelper = new DatabaseHelper(getContext());
+    BookDao bd = new BookDao(dbHelper);
+    NoteDao nd = new NoteDao(dbHelper);
+
+    List<Long> bookIdsCurrShelf = bd.getAllBookIdsForShelf(shelfId);
+    List<Author> authorsCurrBook = new ArrayList<>();
+    List<Long> notesCurrBook = new ArrayList<>();
+    String allNotesCurrBook = "";
+    String authorNamesCurrBook = "";
+
+    //for each book in the current shelf
+    for (int i = 0; i < bookIdsCurrShelf.size(); i++) {
+      Book currBook;
+      Long currBookId = bookIdsCurrShelf.get(i);
+      authorsCurrBook = bd.getAllAuthorsForBook(currBookId);
+      notesCurrBook = nd.getAllNoteIdsForBook(currBookId);
+      currBook = bd.findById(currBookId);
+
+      /*
+      get the notes for the current book
+      and save the bib content in one string
+      */
+      for (int k = 0; k < notesCurrBook.size(); k++) {
+        String noteTextCurrBook = nd.findTextById(notesCurrBook.get(k));
+        allNotesCurrBook +=  "annote={" + noteTextCurrBook + "}," + '\n';
+      }
+
+      /*
+      get author's first and last name and include
+      the needed book data in a bib format
+      */
+      for (int u = 0; u < authorsCurrBook.size(); u++) {
+        if (authorsCurrBook.size() > 1) {
+          authorNamesCurrBook = authorNamesCurrBook + " and "
+              + authorsCurrBook.get(u).getFirstName()
+              + " " + authorsCurrBook.get(u).getLastName();
+        } else {
+          authorNamesCurrBook = authorsCurrBook.get(u).getFirstName()
+              + " " + authorsCurrBook.get(u).getLastName();
+        }
+      }
+
+      //remove the last "and" in case of more than one author
+      if (authorNamesCurrBook.contains("and")) {
+        authorNamesCurrBook = authorNamesCurrBook.substring(0,
+            authorNamesCurrBook.length() - 5);
+      }
+
+      String bookTitle = currBook.getTitle();
+      bookTitle = bookTitle.replaceAll("\\s+", "");
+
+      bibExportContent = bibExportContent
+          + "@book{" + bookTitle + currBook.getPubYear() + "," + '\n'
+          + "isbn={" + currBook.getIsbn() + "}," + '\n'
+          + "author={" + authorNamesCurrBook + "}," + '\n'
+          + "title={" + currBook.getTitle() + "}," + '\n'
+          + "publisher={" + currBook.getPublisher() + "}," + '\n'
+          + "edition={" + currBook.getEdition() + "}," + '\n'
+          + allNotesCurrBook + "," + '\n'
```

Comment:
Commas and linebreaks are added per note before, so adding them here again makes the file incorrect :)

```suggestion
          + allNotesCurrBook + 
```